Networking/NetworkingTests/Remote/ProductVariationsRemoteTests.swift

```diff
@@ -117,6 +117,55 @@ final class ProductVariationsRemoteTests: XCTestCase {
         wait(for: [expectation], timeout: Constants.expectationTimeout)
     }
 
+    // MARK: - Load single product variation tests
+
+    /// Verifies that loadProductVariation properly parses the `product-variation` sample response.
+    ///
+    func test_load_single_ProductVariation_properly_returns_parsed_ProductVariation() throws {
+        // Given
+        let remote = ProductVariationsRemote(network: network)
+        let sampleProductVariationID: Int64 = 1275
+        network.simulateResponse(requestUrlSuffix: "products/\(sampleProductID)/variations/\(sampleProductVariationID)", filename: "product-variation")
+
+        // When
+        var resultMaybe: Result<ProductVariation, Error>?
+        waitForExpectation { exp in
+            remote.loadProductVariation(for: sampleSiteID, productID: sampleProductID, variationID: sampleProductVariationID) { aResult in
+                resultMaybe = aResult
+                exp.fulfill()
+            }
+        }
+
+        // Then
+        let result = try XCTUnwrap(resultMaybe)
+        XCTAssertTrue(result.isSuccess)
+
+        let productVariation = try result.get()
+        XCTAssertEqual(productVariation.productID, sampleProductID)
+        XCTAssertEqual(productVariation.productVariationID, sampleProductVariationID)
+    }
+
+    /// Verifies that loadProductVariation properly relays any Networking Layer errors.
+    ///
+    func test_load_single_ProductVariation_properly_relays_netwoking_errors() throws {
+        // Given
+        let remote = ProductVariationsRemote(network: network)
+        let sampleProductVariationID: Int64 = 2783
+
+        // When
+        var resultMaybe: Result<ProductVariation, Error>?
+        waitForExpectation { exp in
+            remote.loadProductVariation(for: sampleSiteID, productID: sampleProductID, variationID: sampleProductVariationID) { aResult in
+                resultMaybe = aResult
+                exp.fulfill()
+            }
+        }
+
+        // Then
+        let result = try XCTUnwrap(resultMaybe)
+        XCTAssertTrue(result.isFailure)
```

Comment:
We can now also use [`waitFor`](https://github.com/woocommerce/woocommerce-ios/blob/fe221b6ce87d5018a8ce21d5547859ddd6e14639/TestKit/Sources/TestKit/XCTestCase%2BWait.swift#L58-L95) so we can have an immutable variable. 

```suggestion
        // When
        let result = try waitFor { promise in
            remote.loadProductVariation(for: self.sampleSiteID, productID: self.sampleProductID, variationID: sampleProductVariationID) { result in
                promise(result)
            }
        }

        // Then
        XCTAssertTrue(result.isFailure)
```

Reply:
thanks for the suggestion! this `waitFor` helper certainly simplifies a bunch of things, no need to try unwrapping to the non-optional value anymore. I updated all the usage in this PR in https://github.com/woocommerce/woocommerce-ios/pull/2915/commits/26e5adeae232736e45eb0e54e0409e259a190c16